generator/src/test/java/com/scottlogic/deg/generator/restrictions/SetRestrictionsTests.java

```diff
@@ -1,122 +1,199 @@
+        }
 
-    @BeforeEach
-    private void beforeEach() {
-        this.firstInputWhitelist = null;
-        this.firstInputBlacklist = null;
-        this.secondInputWhitelist = null;
-        this.secondInputBlacklist = null;
-        this.actualOutput = null;
+        @Test
+        void cannotMergeWhitelistsWithNoCommonItems() {
+            expectUnmergeable(
+                SetRestrictions.fromWhitelist(set(1, 2)),
+                SetRestrictions.fromWhitelist(set(3, 4)));
+        }
     }
 
-    private Set<Object> firstInputWhitelist = null;
-    private Set<Object> firstInputBlacklist = null;
-    private Set<Object> secondInputWhitelist = null;
-    private Set<Object> secondInputBlacklist = null;
+    @Nested
+    class equality {
+        @Test
+        void isNotEqualToNull() {
+            assertNotEqual(
+                SetRestrictions.fromWhitelist(set("Test")),
+                null);
+        }
 
-    private MergeResult<SetRestrictions> actualOutput = null;
+        @Test
+        void isNotEqualToNonSetRestrictionsObject() {
+            assertNotEqual(
+                SetRestrictions.fromWhitelist(set("Test")),
+                "String");
+        }
 
-    private MergeResult<SetRestrictions> getActualOutput() {
-        if (actualOutput == null) {
-            actualOutput = SetRestrictions.merge(
-                new SetRestrictions(firstInputWhitelist, firstInputBlacklist),
-                new SetRestrictions(secondInputWhitelist, secondInputBlacklist));
+        @Test
+        void twoEmptyWhitelistsAreEqual() {
+            assertEqual(
+                SetRestrictions.fromWhitelist(Collections.emptySet()),
+                SetRestrictions.fromWhitelist(Collections.emptySet()));
         }
 
-        return actualOutput;
-    }
+        @Test
+        void twoEmptyBlacklistsAreEqual() {
+            assertEqual(
+                SetRestrictions.fromBlacklist(Collections.emptySet()),
+                SetRestrictions.fromBlacklist(Collections.emptySet()));
+        }
 
-    private void givenFirstInputWhitelist(Object... expectedValues) {
-        firstInputWhitelist = new HashSet<>(Arrays.asList(expectedValues));
-    }
-    private void givenFirstInputBlacklist(Object... expectedValues) {
-        firstInputBlacklist = new HashSet<>(Arrays.asList(expectedValues));
-    }
-    private void givenSecondInputWhitelist(Object... expectedValues) {
-        secondInputWhitelist = new HashSet<>(Arrays.asList(expectedValues));
-    }
-    private void givenSecondInputBlacklist(Object... expectedValues) {
-        secondInputBlacklist = new HashSet<>(Arrays.asList(expectedValues));
+        @Test
+        void twoDifferentBlacklistsAreNotEqual() {
+            assertNotEqual( // same type
+                SetRestrictions.fromBlacklist(set("Test")),
+                SetRestrictions.fromBlacklist(set("Parrot")));
+
+            assertNotEqual( // different type
```

Comment:
You're testing two things at once here.

Reply:
I'd say I'm testing one expectation in two ways. It'd be excessively noisy to have them as completely separate tests, since they're barely meaningfully different. I think the ideal would be to use a parameterised test, but the RoI on me looking up how those work in JUnit didn't seem compelling :)